hoard/client.py

```diff
@@ -61,23 +62,41 @@ def __init__(self, source_url: str, format: str, set: str = None) -> None:
         self.ids: Optional[Iterator] = None
         client = Sickle(self.source_url)
         self.client = client
+        session = requests.Session()
+        self.session = session
 
     def __iter__(self) -> Iterator[str]:
         return self
 
     def __next__(self) -> str:
         if self.ids is None:
             self.ids = self.fetch_ids()
-        client = self.client
+        session = self.session
         while True:
-            id = next(self.ids)
-            record = client.GetRecord(
-                identifier=id.identifier, metadataPrefix=self.format
-            )
-            if record.deleted:
-                continue
+            header_xml = next(self.ids)
+            parsed_header = ET.fromstring(header_xml.raw)
+            namespace = {"oai": "http://www.openarchives.org/OAI/2.0/"}
+            id_elem = parsed_header.find("oai:identifier", namespace)
+            if id_elem is None:
+                raise Exception("No OAI identifier found")
             else:
-                return record
+                id = id_elem.text
+            params = {
+                "verb": "GetRecord",
+                "metadataPrefix": self.format,
+                "identifier": id,
+            }
+            record = session.get(self.source_url, params=params).text
+            parsed_record = ET.fromstring(record)
+            for elem in [
```

Comment:
This bit could all be replaced with a pretty simple XPath expression:
```python
deleted = parsed_record.find(".//oai:header[@status='deleted']", namespaces=namespace)
if deleted:
    continue
else:
    return record
```

Reply:
I figure there was a cleaner way to do this but I laughed at how much simpler this is, thanks!